pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/JSONParser.groovy

```diff
@@ -419,19 +420,26 @@ class JSONParser implements Parser {
     }
 
     public @CheckForNull ModelASTValue parseValue(JsonTree o) {
+        ModelASTValue val = null
         if (o.node.get("isLiteral").asBoolean()) {
             if (o.node.get("value").isBoolean()) {
-                return ModelASTValue.fromConstant(o.node.get("value").booleanValue(), o)
+                val = ModelASTValue.fromConstant(o.node.get("value").booleanValue(), o)
             } else if (o.node.get("value").isNumber()) {
-                return ModelASTValue.fromConstant(o.node.get("value").numberValue(), o)
+                val =  ModelASTValue.fromConstant(o.node.get("value").numberValue(), o)
             } else if (o.node.get("value").isTextual()) {
-                return ModelASTValue.fromConstant(o.node.get("value").textValue(), o)
+                val = ModelASTValue.fromConstant(o.node.get("value").textValue(), o)
             } else {
-                return ModelASTValue.fromConstant(o.node.get("value").textValue(), o)
+                val = ModelASTValue.fromConstant(o.node.get("value").textValue(), o)
             }
         } else {
-            return ModelASTValue.fromGString(o.node.get("value").textValue(), o)
+            val = ModelASTValue.fromGString(o.node.get("value").textValue(), o)
         }
+
+        String valGroovy = val?.toGroovy()
+        if (valGroovy != null && valGroovy != StringEscapeUtils.escapeJava(valGroovy)) {
```

Comment:
Well, turns out that didn't actually work the way I thought it did anyway. New commit(s) incoming.